Lib/test/test_embed.py

```diff
@@ -346,7 +342,8 @@ class InitConfigTests(EmbeddingTestsMixin, unittest.TestCase):
         'prefix',
         'pycache_prefix',
         'warnoptions',
```

Comment:
You switch `warnoptions` over to using the new `COPY_LIST` macro below, as well as updating it to be read from the `sys` module when needed. Did you mean to also remove it from this list?

Reply:
And that's what I get for reviewing the code before reading the author's initial comments on the review - you already answered that question (the `warnoptions` here are the warning options as specified in the initial configuration, while the "live" state is always the copy in the sys module).

Reply:
warnoptions comes from DEFAULT_CORE_CONFIG: warnoptions=[]. Can it can be replaced in other tests like: warnoptions=['default', 'error::ResourceWarning']. COPY_MAIN_CONFIG ensures that core and main configuration contain exactly the same value, it's just to not repeat the expected value twice.

Spoiler(!): I'm now experimenting a change to avoid redundancy between core and main configuration! Stay tuned.